homeassistant/components/egps/__init__.py

```diff
@@ -0,0 +1,42 @@
+"""Energenie Power-Strip (EGPS) integration."""
+
+from pyegps import PowerStripUSB, get_device
+from pyegps.exceptions import MissingLibrary, UsbError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, IntegrationError
+
+from .const import CONF_DEVICE_API_ID, DOMAIN
+
+PLATFORMS = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Energenie Powerstrip."""
+    try:
+        powerstrip: PowerStripUSB | None = get_device(entry.data[CONF_DEVICE_API_ID])
+
+    except (MissingLibrary, UsbError) as ex:
+        raise IntegrationError("Can't access usb devices.") from ex
+
+    if powerstrip is None:
+        raise ConfigEntryNotReady
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = powerstrip
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        powerstrip = hass.data[DOMAIN].pop(entry.entry_id)
+        powerstrip.release()
```

Comment:
can this throw exceptions?

Reply:
releasing the device should be fine even when it is not physically connected any more

Reply:
no other exceptions possible? I'm just being careful here as an exception here would prevent you from reloading the config entry and/or delay homeassistant shutdowns

Reply:
the exception handling is already done inside the pyusb library:
```python
def release_all_interfaces(self, device):
        claimed = copy.copy(self._claimed_intf)
        for i in claimed:
            try:
                self.managed_release_interface(device, i)
            except USBError:
                # Ignore errors when releasing the interfaces
                # When the device is disconnected, the call may fail
                pass
``` 
Restarting HA with the usb device disconnected works.

The only other thing that theoretically can raise is a key error when removing the device from the dictionary. Even though this should not happen, I will add a checking to the pyegps api library to make sure to prevent this.

This means, we need to update the dependency version of pyegps then.

Reply:
done.